### DISCUSSION TITLE
hss-payload-generator for Windoze

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#344] [HSS and U-Boot]
Hi have a Windows based customer using the PolarFire SoC and that includes HSS.   They need the hss-payload-generator executable for Windows.    For Linux we provide the means to build it no problem but for Windows the Make infrastructure is all Linux centric.
Has anyone gone down this path of building the hss-payload-generator for Windows?

Thx!
Dave

**Comment by rockoguido** [comment]
ok answering myself, going to pursue the use of MSYS2 on Windoze to build hss-payload-generator.    looks like a viable option

**Reply by rockoguido** [reply]
lots of dialog with myself here.     having an issue with MSYS2, unable to find the libelf or libelf-dev package, hence having compilation issues when trying to build
turns out you can install libelf via 
pacman -S mingw-w64-x86_64-libelf

to get GCC..
pacman -S mingw-w64-x86_64-gcc

What is driving this is this error when building hss-payload-generator in MSYS2

 CC        /home/C68141/hart-software-services/tools/hss-payload-generator/main.o
 CC        /home/C68141/hart-software-services/tools/hss-payload-generator/yaml_parser.o
 CC        /home/C68141/hart-software-services/tools/hss-payload-generator/blob_handler.o
blob_handler.c:36:10: fatal error: gelf.h: No such file or directory
   36 | #include <gelf.h>
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:88: /home/C68141/hart-software-services/tools/hss-payload-generator/blob_handler.o] Error 1

we do have...via the root partition..
cd /
find . -name gelf.h
./mingw64/include/gelf.h
./mingw64/include/libelf/gelf.h

of course don't want to hardwire the include path to gelf.h, other issues will most like percolate to the top.   think I need to sort out how to get the libelf-dev package found and downloaded

**Comment by griffini** [comment]
Windows (and Linux) binaries for the HSS Payload Generator are available at https://github.com/polarfire-soc/hart-software-services/releases/download/v2023.02/hss-payload-generator.zip

The [HSS Payload Generator README](https://github.com/polarfire-soc/hart-software-services/blob/master/tools/hss-payload-generator/README.md) provides some instructions on building for MSYS2. Specifically, to install libyaml and libelf in MSYS2, use:

````
    $ pacman -S libyaml libyaml-devel libelf libelf-devel zlib zlib-devel openssl-devel
````

**Comment by rockoguido** [comment]
Hi

Thanks so much for the link to the prebuilt executables, very handy!!

I did before attempt to go down the path of setting up the environment via pacman as you describe below..
Have 2 issues, getting access to libelf and libelf-devel.
I did manage to install libelf via the following command
                pacman -S mingw-w64-x86_64-libelf

However still not able to obtain libelf-devel, just get a target not found error.

Just wanted to note that

Thx again
Dave

Dave Campanella
Principal Embedded Solutions Engineer / FAE
Microchip Technology Inc.
165 Technology Dr
Irvine, CA 92618
+1 925 989 0569 (mobile)
+1 760 304 0172 (office)
***@***.******@***.***>

[Text  Description automatically generated]
<http://www.microchip.com/emasters>Visit our new learning site at www.microchip.com/mu<http://www.microchip.com/mu>

**Comment by griffini** [comment]
Interesting.  I just tried from a clean setup and I get the same error.  My pre-existing setup already has libelf-devel installed so it works fine.

This seems to be an issue with MSYS2.  As a workaround, you can download the libelf packages directly from the following URLs and install manually:

https://repo.msys2.org/msys/x86_64/libelf-0.8.13-2-x86_64.pkg.tar.xz
https://repo.msys2.org/msys/x86_64/libelf-devel-0.8.13-2-x86_64.pkg.tar.xz

To install manually, run the following from MSYS2, in the directory where you downloaded those files:

````
pacman -U libelf-0.8.13-2-x86_64.pkg.tar.xz
pacman -U libelf-devel-0.8.13-2-x86_64.pkg.tar.xz
````

After this, you should be able to build the payload generator okay.

**Comment by rockoguido** [comment]
Hi!

Yes that works great, got the 2 packages libelf and libelf-devel installed and now can build the payload generator without an issue.   Do appreciate the help.

A higher level question I have is one a user has the HSS PAYLOAD GENERATOR built is there ever a reason to rebuild it?

It seems all of the target specific configuration is managed via the YAML file and then of course the .elf and .bin files are input also to generate a .bin image.

Thx again!

Dave Campanella
Principal Embedded Solutions Engineer / FAE
Microchip Technology Inc.
165 Technology Dr
Irvine, CA 92618
+1 925 989 0569 (mobile)
+1 760 304 0172 (office)
***@***.******@***.***>

[Text  Description automatically generated]
<http://www.microchip.com/emasters>Visit our new learning site at www.microchip.com/mu<http://www.microchip.com/mu>
